### PR TITLE
skip overload actions for static listeners

### LINKED ISSUE #52663: Allow to disable overload manager for health checks and metrics
**Describe the feature request**

In [Envoy 1.31](https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.31/v1.31.0), a new feature was added to the listener called `bypass_overload_manager`:

> listener: Added [bypass_overload_manager](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener.proto#envoy-v3-api-field-config-listener-v3-listener-bypass-overload-manager) to bypass the overload manager for a listener. When set to true, the listener will not be subject to overload protection.

This feature is useful to be enabled for health check and metrics listeners, so that they can bypass the overload manager.

For example, the overload manager might be configured to drop any new connections or HTTP requests when the memory reaches 95%. 

However, this affects both health checks and metrics, leading to a problem that the metrics will not be exported, or the proxy might be killed, because it won't respond to the health check.

As for now, the health checks and metrics listeners are configured using [bootstrap configuration](https://github.com/istio/istio/blob/master/tools/packaging/common/envoy_bootstrap.json). This bootstrap configuration might be overridden, though, I am not sure if we can override these listeners, and how complex it is to do this.

My proposal is to add an env variable, similar to `ISTIO_DUAL_STACK`, that will allow to configure the `bypass_overload_manager` for the static listeners in the bootstrap file.

**Describe alternatives you've considered**

An alternative is to override the bootstrap configuration. However, I am not sure how custom bootstrap configuration is merged with the default one. It seems to me that adding a new property to the bootstrap configuration is pretty straight-forward, but changing existing values is not. 

Moreover, the listeners configuration is provided as an array, which might be more difficult to override. 

Lastly, overriding the bootstrap configuration is more error-prone between Istio versions. Having an explicit option seem to be a better choice.

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

Related issues - #41859

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/52663
https://github.com/istio/istio/issues/41859

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: manifests/helm-profiles/compatibility-version-1.23.yaml [hunk 1]
```diff
@@ -9,4 +9,5 @@ meshConfig:
   defaultConfig:
     proxyMetadata:
       # 1.24 behaviour changes
-      ENABLE_DEFERRED_STATS_CREATION: "false"  
+      ENABLE_DEFERRED_STATS_CREATION: "false"
+      BYPASS_OVERLOAD_MANAGER_FOR_STATIC_LISTENERS: "false"
```
Comment (MorrisLaw) [inline manifests/helm-profiles/compatibility-version-1.23.yaml:13]: For my context and understanding, what's the significance of adding this env variable here as opposed to in the pilot.env section?

Comment (ramaraochavali) [reply manifests/helm-profiles/compatibility-version-1.23.yaml:13]: This is relevant to Pilot Agent - Not Pilot.

**Review by hzxuzhonghu — COMMENTED** [review]
How does envoy know how much cpu / memory it is configurred?

File: tools/packaging/common/envoy_bootstrap.json [hunk 2]
```diff
@@ -550,6 +550,10 @@
             "port_value": {{ .envoy_prometheus_port }}
           }
         },
+        {{- if .bypass_overload_manager}}
+        "ignore_global_conn_limit": true,
+        "bypass_overload_manager": true,
+        {{ end}}
```
Comment (hzxuzhonghu) [inline tools/packaging/common/envoy_bootstrap.json:556]: And you did not set `overload_manager`

**Comment by ramaraochavali** [comment]
> How does envoy know how much cpu / memory it is configurred?

We have to configure it via configuration https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/overload_manager/overload_manager#config-overload-manager - see as an example

**Comment by hzxuzhonghu** [comment]
But i donot see the config in this patch?

File: tools/packaging/common/envoy_bootstrap.json (hunk 2 above)
Comment (ramaraochavali) [reply tools/packaging/common/envoy_bootstrap.json:556]: I think this is to allow bypassing overload manager if/when it is configured. In this PR I am not configuring it. If folks use overload manager via https://github.com/istio/istio/issues/14366#issuecomment-548141596 - they get this benefit.

**Comment by ramaraochavali** [comment]
> But i donot see the config in this patch?

No. This just ensures, if overload config is enabled by some means(bootstrap merge for example), the static listeners are not counted under overload actions. This PR does not enable overload manager.

**Review by keithmattix — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]
Need to emphasize this still need customizing bootstrap config https://github.com/istio/istio/issues/14366#issuecomment-548141596

**Pull request merged**